handlers/secrets.go

```diff
@@ -82,7 +82,13 @@ func UpdateSecrets(request requests.CreateFunctionRequest, deployment *v1beta1.D
 			},
 		}
 
-		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, projectedSecrets)
+		existingVolumes := deployment.Spec.Template.Spec.Volumes
+		for i, v := range existingVolumes {
+			if v.Name == volumeName {
+				existingVolumes = append(existingVolumes[:i], existingVolumes[i+1:]...)
```

Comment:
Maybe I'm missing something.. why can't we use the more readable version?

```
existingVolumes = append(existingVolumes, v)
```

Reply:
Please could you change the style so that it's not destructive? I think readability is a concern too.

Reply:
@alexellis its removing the element not appending it at this point, for example:
```
package main

import (
	"fmt"
)

func main() {
	s := []string{"a", "b", "c", "d", "e", "f"}
	fmt.Printf("original slice      : %s\n", s)
	for i, e := range s {
		if e == "c" {
			s = append(s[:i], s[i+1:]...)
		}
	}
	fmt.Printf("remove c from slice : %s\n", s)
}
```
https://play.golang.org/p/Yg6xixVtX44

Reply:
@LucasRoesler can  you `break` when you match the volume?

Reply:
@alexellis non-destructive alternative? (`Volume` needs pointed at the correct struct below)
```
existingVolumes := []Volume{}
for _, v := range  deployment.Spec.Template.Spec.Volumes{
    if v.Name != volumeName {
        existingVolumes = append(existingVolumes, v)
    }
}
```

Reply:
@johnmccabe break definitely makes sense, if the Deployment is for an existing function coming from k8s, then the volume names are already unique and if it exists, it should only exist once.  I guess the question is, could there be a situation in which it was already duplicated in the volumes list? It would obviously be a broken deployment, but this code would cover that edge case and ensure that the volume really is only listed once.

Reply:
Ok, easy enough, I will have to it tomorrow because I just left the house for the evening.